src/controlobjectscript.cpp

```diff
@@ -0,0 +1,45 @@
+#include <QApplication>
+#include <QtDebug>
+
+#include "controlobjectscript.h"
+
+ControlObjectScript::ControlObjectScript(const ConfigKey& key, QObject* pParent)
+        : ControlObjectSlave(key, pParent) {
+}
+
+ControlObjectScript::~ControlObjectScript() {
+    //qDebug() << "ControlObjectScript::~ControlObjectSlave()";
+}
+
+bool ControlObjectScript::connectScriptFunction(
+        const ControllerEngineConnection& conn) {
+    m_connectedScriptFunction.append(conn);
+    connect(m_pControl.data(), SIGNAL(valueChanged(double, QObject*)),
+            this, SLOT(slotValueChanged(double,QObject*)),
+            static_cast<Qt::ConnectionType>(Qt::QueuedConnection |
+                                            Qt::UniqueConnection));
+    connect(this, SIGNAL(trigger(double, QObject*)),
+            this, SLOT(slotValueChanged(double,QObject*)),
+            static_cast<Qt::ConnectionType>(Qt::QueuedConnection |
+                                            Qt::UniqueConnection));
+    return true;
+}
+
+bool ControlObjectScript::disconnectScriptFunction(
+        const ControllerEngineConnection& conn) {
+    return m_connectedScriptFunction.removeAll(conn) > 0;
```

Comment:
The code is not symmetric. In connectScriptFunction() you are actually connecting something, but here you only clear a list. This should at least be explained. I don't get it when reading this code.

Better: Do connects and disconnects at the same abstraction level, i.e. in corresponding functions.